Fix bugs in setup-platform; update platform_config and volttron submodule

bonicim opened the pull request:
What is the change?
* Fix script errors for installing agents
* Add install web-based packages on bind-web-address
* Upate platform_config.yml

Why is it needed?
* To allow the Dockerfile to successfully build an image when docker run or docker-compose is used to create a container.

How tested?
I ran the following docker run command to build the container. 

```
$ docker run -e LOCAL_USER_ID=$UID -e CONFIG=/home/volttron/configs \ 
-v $HOME/develop/volttron-docker/configs:/home/volttron/configs \
-v $HOME/develop/volttron-docker/platform_config.yml:/platform_config.yml \ 
-p 8080:8080 \
-it volt_develop 
```

 Note that for the volttron submodule, I used the latest from develop; the image also requires an update to requirements.txt, which is noted in this PR https://github.com/VOLTTRON/volttron/pull/2468

bonicim commented:
The docker created from this PR uses the volttron submodule at https://github.com/bonicim/volttron/commit/0b40f54a3ee41c213f0c67bd4647867ba0ef142b. That particular version fixes the rabbit installation script so that it works on debian buster.

bonicim commented:
Added `build` to docker-compose.yml so that it always uses the local Dockerfile instead of going to Dockerhub.

craigpnnl commented on README.md:
````diff
@@ -2,11 +2,66 @@
+$ ./docker_install_ubuntu.sh
+```
+
+## Gotcha
````
This is not a Gotcha....it is a just a way of making it repeatable at a set point in time.  This is not necessary unless you want to build it locally.  Which I believe is why I didn't have the build in the docker-compose file originally.

bonicim replied:
I understand now. Will rename and move to different section.

craigpnnl commented on README.md:
````diff
@@ -19,6 +19,23 @@ If you need to install docker and/or docker-compose, you can use the script in t
+When you initially clone this repo, the 'volttron' directory is empty. This directory contains the volttron codebase used to create the volttron platform. Before creating the container, you must pull in volttron from the official volttron repo using the following git command:
+
+```bash
+# Ensure that you are in the `volttron directory` 
````
We don't need to be in the volttron directory to run this...because the module is in .gitmodules.  And that has the relative path form the root of the repository.

bonicim replied:
> Instead of a Gotcha please add that to a building image locally section. For the most part what we want is to not have the user have to do that and just use the image that is out on dockerhub.
> 
> Including a building/building the latest code branch section would emphasis this.

Thanks for the clarification. I've been building and testing locally for so long that I didn't realize this step is not necessary to run docker-compose. I just emptied that folder and ran `docker-compose up` and it runs fine. I already created a local development section, so I'll move this to there.

craig8 reviewed (changes requested):
Instead of a Gotcha please add that to a building image locally section.  For the most part what we want is to not have the user have to do that and just use the image that is out on dockerhub.

Including a building/building the latest code branch section would emphasis this.

craigpnnl commented:
I am getting the following error building the image.


 ---> Running in 9552088a39a1
                                                                                                        │+ apt-get update
                                                                                                        │Get:1 http://deb.debian.org/debian buster InRelease [121 kB]
                                                                                                        │Get:2 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
                                                                                                        │Get:3 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]
                                                                                                        │Get:4 http://deb.debian.org/debian buster/main amd64 Packages [7906 kB]
                                                                                                        │Reading package lists...
                                                                                                        │E: Release file for http://security.debian.org/debian-security/dists/buster/updates/InRelease is not val                                                                                                        │id yet (invalid for another 1d 1h 3min 55s). Updates for this repository will not be applied.
                                                                                                        │E: Release file for http://deb.debian.org/debian/dists/buster-updates/InRelease is not valid yet (invali                                                                                                        │d for another 1d 19h 53min 36s). Updates for this repository will not be applied.
                                                                                                        │ERROR: Service 'volttron' failed to build: The command 'bash -c set -eux; apt-get update; apt-get instal                                                                                                        │l -y --no-install-recommends     procps     gosu     vim     tree     build-essential     python3-dev                                                                                                           │  python3-pip     python3-setuptools     python3-wheel     openssl     libssl-dev     libevent-dev     g                                                                                                        │it     gnupg     dirmngr     apt-transport-https     wget     curl     ca-certificates     libffi-dev' r                                                                                                        │eturned a non-zero code: 100

craigpnnl commented:
The above issue is solved because I had suspend drift happening with the virtual machine!

craigpnnl commented on platform_config.yml:
```diff
@@ -40,8 +40,17 @@ agents:
+      fake.csv:
+        file: $CONFIG/bacnet_example_config.csv
+        type: --csv
+      devices/PNNL/ANNEX:
```
Remove all reference to PNNL in the files....

craigpnnl commented:
When looking into things, I noticed that there wasn't any graphs available in vc.  After going into the container and looking at the historian database I noticed there isn't any data in it.

To test this please 

apt install sqlite3

cd into the ~/.volttron/agents/<agent_uuid>/sqlhistorianagent-3.7.0/data

sqlite3 historian.sqlite

select * from data;

craigpnnl commented:
After some inspection it seems that the master driver files don't get added to the config store, as they are not included in the files that get copied to the host.  the historian.config and listener.config do however not any of the master driver configs did.

bonicim commented:
> When looking into things, I noticed that there wasn't any graphs available in vc. After going into the container and looking at the historian database I noticed there isn't any data in it.
> 
> To test this please
> 
> apt install sqlite3
> 
> cd into the ~/.volttron/agents/<agent_uuid>/sqlhistorianagent-3.7.0/data
> 
> sqlite3 historian.sqlite
> 
> select * from data;

Got it. After rebuilding the image and starting the container, I was able to see data: 

<details>
  <summary>output from select * from data;</summary>
2020-10-01T18:09:55.002276+00:00|23|-0.49999999999999967
2020-10-01T18:09:55.002276+00:00|24|-0.49999999999999967
2020-10-01T18:10:00.004368+00:00|1|50.0
2020-10-01T18:10:00.004368+00:00|2|50
2020-10-01T18:10:00.004368+00:00|3|true
2020-10-01T18:10:00.004368+00:00|4|50.0
2020-10-01T18:10:00.004368+00:00|5|50
2020-10-01T18:10:00.004368+00:00|6|true
2020-10-01T18:10:00.004368+00:00|7|50.0
2020-10-01T18:10:00.004368+00:00|8|50
2020-10-01T18:10:00.004368+00:00|9|true
2020-10-01T18:10:00.004368+00:00|10|false
2020-10-01T18:10:00.004368+00:00|11|0.0
2020-10-01T18:10:00.004368+00:00|12|10.0
2020-10-01T18:10:00.004368+00:00|13|20
2020-10-01T18:10:00.004368+00:00|14|true
2020-10-01T18:10:00.004368+00:00|15|10.0
2020-10-01T18:10:00.004368+00:00|16|20
2020-10-01T18:10:00.004368+00:00|17|true
2020-10-01T18:10:00.004368+00:00|18|10.0
2020-10-01T18:10:00.004368+00:00|19|20
2020-10-01T18:10:00.004368+00:00|20|true
2020-10-01T18:10:00.004368+00:00|21|0
2020-10-01T18:10:00.004368+00:00|22|0
2020-10-01T18:10:00.004368+00:00|23|0.0
2020-10-01T18:10:00.004368+00:00|24|0.0
</details>

bonicim commented:
> After some inspection it seems that the master driver files don't get added to the config store, as they are not included in the files that get copied to the host. the historian.config and listener.config do however not any of the master driver configs did.

Makes sense. I corrected the platform_config.yml to the following based on meeting with Shewtha. 

```yml
  platform.driver:
    source: $VOLTTRON_ROOT/services/core/MasterDriverAgent
    config_store:
      fake.csv:
        file: $VOLTTRON_ROOT/examples/configurations/drivers/fake.csv
        type: --csv
      devices/fake-campus/fake-building/fake-device:
        file: $VOLTTRON_ROOT/examples/configurations/drivers/fake.config
```

bonicim commented:
Addressed feedback from @craig8. After creating the container using `docker-compose up`, I inspected the web interface to check for charts and checked for data in the historian by running `select * from data` in the container.

bonicim commented on docker-compose.yml:
```diff
@@ -2,10 +2,10 @@ version: '3'
   volttron:
     hostname: vc
-    image: volttron/volttron
+    image: bonicim/volttron-docker-dev:test
```
I built the image using the Dockerfile in this repo and pushed to my personal dockerhub account. For now, using this image to run docker-compose.

bonicim commented on platform_config.yml:
```diff
@@ -29,19 +29,25 @@ agents:
+      devices/fake-campus/fake-building/fake-device:
+        file: $VOLTTRON_ROOT/examples/configurations/drivers/fake.config
 
   platform.agent:
```
Since agents are installed sequentially,  I realized that the platform agent needs to be built after the voltton central agent is created because platform agent needs the `bacpypes` dependency, which it doesn't install on its own. However, volttron central agent does install that dependency.
